src/main/java/com/springboot/star_wars_character_popularity/app/service/serviceImpl/CharacterImportServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.springboot.star_wars_character_popularity.app.service.serviceImpl;
+
+import com.springboot.star_wars_character_popularity.app.help.CharacterPage;
+import com.springboot.star_wars_character_popularity.app.model.Character;
+import com.springboot.star_wars_character_popularity.app.model.Planet;
+import com.springboot.star_wars_character_popularity.app.repository.CharacterRepository;
+import com.springboot.star_wars_character_popularity.app.service.CharacterImportService;
+import com.springboot.star_wars_character_popularity.app.service.PlanetImportService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Service
+public class CharacterImportServiceImpl implements CharacterImportService {
+
+    @Value("${planet.uri}")
+    String planetsUri;
+
+    PlanetImportService planetImportServiceImpl;
+
+    private CharacterRepository characterRepository;
+
+    public CharacterImportServiceImpl(CharacterRepository characterRepository, PlanetImportServiceImpl planetImportServiceImpl){
+        this.characterRepository=characterRepository;
+        this.planetImportServiceImpl=planetImportServiceImpl;
+    }
+
+    @Override
+    public void fetchCharactersOptimized() {
+
+        List<Planet> allAvailablePlanets = planetImportServiceImpl.getListOfAllPlanets();
+
+        RestTemplate restTemplateCharacters = new RestTemplate();
+        CharacterPage characterPage = restTemplateCharacters.getForObject("https://swapi.dev/api/people", CharacterPage.class);
```

Comment:
if you are already using externalized properties, make one for "https://swapi.dev/api/people"

Reply:
Done.